### PR TITLE
Csb 1588563466617

### PR DESCRIPTION
**Pull request by jalvarez2016**

### CODE REVIEW

**Comment by jalvarez2016** [comment]
merged onto main branch

File: .codesandbox/workspace.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        {
+          "id": "codesandbox.problems"
+        },
+        {
+          "id": "codesandbox.react-devtools"
+        }
+      ]
+    }
+  ]
+}
```
Comment (jtessler) [inline .codesandbox/workspace.json:27]: Please remove this file from this pull request. We don't need to keep Code Sandbox settings around. :smile:

File: src/GameComponent.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
-import React from 'react';
-import firebase from 'firebase';
-import gameData from './gameData.js';
+import React from "react";
```
Comment (jtessler) [inline src/GameComponent.js:1]: Please revert changes to this base class. Looks like an auto-formatter, but you shouldn't need to touch this file in this pull request.

File: src/gameData.js [hunk 3]
```diff
@@ -151,7 +152,15 @@ const gameData = {
...
-
+  SpaceInvaders: {
+    title: "Space Invaders",
+    authors: "Jason Alvarez, Patrick Cooney, David ",
+    description:
+      "Are you going to let Aliens invade your perfect dictatorship? Of course not! Defend your glorious planet.",
+    minUsers: 1,
+    maxUsers: 10,
+    component: SpaceInvaders
+  }
```
Comment (jtessler) [inline src/gameData.js:163]: Please only include this entry and the import line in this pull request (and not the other edits). Looks like the auto-formatter in Code Sandbox hates my style. :stuck_out_tongue:

File: src/games/SpaceInvaders/SpaceInvaders.jsx [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+        left: 300,
+        top: 400,
+        health: 3,
+        shots: []
+      },
+      score: 0
+    };
+
+    this.getSessionDatabaseRef().set({
+      playerNames: UserApi.getName(this.getSessionUserIds()),
```
Comment (jtessler) [inline src/games/SpaceInvaders/SpaceInvaders.jsx:36]: It doesn't look like you use `playerNames` anywhere. Delete it?

File: src/games/SpaceInvaders/SpaceInvaders.jsx [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+          ...this.state.playerTwo,
+          shots: newShots
+        },
+        enemies: newEnemies
+      });
+    }
+  };
+
+  enemyFall = num => {
+    // if (this.isCreator()) {
```
Comment (jtessler) [inline src/games/SpaceInvaders/SpaceInvaders.jsx:112]: Why did you remove this `isCreator()` check? This means both player one and two send the same `newEnemies` list to Firebase, which might lead to problems. I think it makes sense for only the game creator to send this data to Firebase.

File: src/games/SpaceInvaders/PlanetHealth.jsx [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+  } else {
+    gameOver();
+  }
+}
+hitz();
+healthDisplay();
+destroyPlanetz();
+//----------------Test2UwU-----------// (Totally Not Adapted From Something Else)
+
+class PlanetHealthBar extends React.Component {
```
Comment (jtessler) [inline src/games/SpaceInvaders/PlanetHealth.jsx:37]: Where is this component used? Can it be deleted?

File: src/games/SpaceInvaders/enemyCollide.jsx [hunk 7]
```diff
@@ -0,0 +1,18 @@
+function enemyCollide(enemy, shots) {
```
Comment (jtessler) [inline src/games/SpaceInvaders/enemyCollide.jsx:1]: Is `enemyCollide(...)` used anywhere? Please remove this file if not.

**Review by jtessler — CHANGES_REQUESTED** [review]
This looks awesome, Jason! Excellent work!

I just have some comments to keep our code as clean as possible and not check in any unused code. Let me know if you need any help with this review. You can reach out to me on Slack!

**Comment by jtessler** [comment]
As discussed offline, I merged this manually from [another repo] in [this commit].

[another repo]:https://github.com/MLGpepe01/SpaceInvaders
[this commit]:https://github.com/jtessler/studio-multiplayer-game/commit/382b4f7f60b91e87ce57440fb5869ee344d90c03

**Pull request closed without merging**